---
title: Experiments
color: green
---

Lorem ipsum

{{% accordion id="ex1" title="Puring gels" %}}

**Puring gels:**

*Why did we choose this experiment?*

For Coomassi staining and western blots gels are needed. The choice of
gel and % of the gel is dependent on the protein.

**Procedure:**

**SDS Gel for 2 Gels:**

Resolving Gel:

| Percentage Gel      | 8%    | 10%   | 15%   |
| ------------------- | ----- | ----- | ----- |
| H2O                 | 4.6   | 4.0   | 2.3   |
| 30% acrylamidmix    | 2.7   | 3.3   | 5.0   |
| 1,5 M Tris (pH 8.8) | 2.5   | 2.5   | 2.5   |
| 10% SDS             | 0.1   | 0.1   | 0.1   |
| 10 % APS            | 0.1   | 0.1   | 0.1   |
| TEMED               | 0.004 | 0.004 | 0.004 |

Stacking gel:

| Percentage Gel    | 5%    |
| ----------------- | ----- |
| H2O               | 2.1   |
| 30% acrylamidmix  | 0.33  |
| 1 M Tris (pH 6.8) | 0.25  |
| 10% SDS           | 0.02  |
| 10 % APS          | 0.02  |
| TEMED             | 0.002 |

**BisTris Gel for 2 Gels:**

Resolving Gel:

| Percentage Gel   | 8%    | 10%   | 12%   |
| ---------------- | ----- | ----- | ----- |
| H2O              | 8.74  | 7.41  | 6.07  |
| 30% acrylamidmix | 5.33  | 6.67  | 8.0   |
| 1,5 M BisTris    | 5.72  | 5.72  | 5.72  |
| 10 % APS         | 0.2   | 0.2   | 0.2   |
| TEMED            | 0.008 | 0.008 | 0.008 |

Stacking gel:

| Percentage Gel   | 5%   |
| ---------------- | ---- |
| H2O              | 7.13 |
| 30% acrylamidmix | 1.43 |
| 1,5 M BisTris    | 1.33 |
| 10 % APS         | 0.1  |
| TEMED            | 0.01 |

{{% /accordion %}}


{{% accordion id="ex2" title="Handling of IDT sequences" %}}

**Handling of IDT sequences:**

**Procedure:**

1.  spin down aliquots

2.  add required volume of TE buffer (20 µl)

3.  vortex thoroughly

4.  incubate on the heating block at 50°C for 20 min

5.  Store at -20°C

{{% /accordion %}}


{{% accordion id="ex3" title="Preparation of all buffers" %}}

**Preparation of all buffers:**


**LB Medium:**

| 10 g/L | Tryptone      |
| ------ | ------------- |
| 5 g/L  | Yeast extract |
| 10 g/L | NaCl          |

  - Sterilize by autoclaving and store at room temperature.

**CC buffer:**

| 10mM   | Hepes            | 2.38 g/L  |
| ------ | ---------------- | --------- |
| 15 mM  | CaCl<sub>2</sub| 2.21 g/L  |
| 55 mM  | MnCl2xH2O        | 10.89 g/L |
| 250 mM | KCl              | 18.64 g/L |

  - Dissolve all components except MnCl2 and adjust the pH to 6.7 with
    KOH. Then add the MnCl2 an filter sterilize the solution over a 0.22
    µm filter

**SOC/SOB medium:**

1.  To 950 mL of deionized H2O, add:

| Tryptone      | 20g  |
| ------------- | ---- |
| Yeast extract | 5g   |
| NaCl          | 0.5g |

1.  SOC medium is identical to SOB medium, except that it contains 20 mM
    glucose. To prepare SOB medium, combine the above ingredients and
    shake until the solutes have dissolved. Add 10 mL of a 250 mM
    solution of KCl. (This solution is made by dissolving 1.86 g of KCl
    in 100 mL of deionized H2O.)

2.  Adjust the pH of the medium to 7.0 with 5 N NaOH (∼0.2 mL). Adjust
    the volume of the solution to 1 L with deionized H2O. Sterilize by
    autoclaving for 20 min at 15 psi (1.05 kg/cm2) on liquid cycle.

3.  Just before use, add 5 mL of a sterile solution of 2 M MgCl2 and 5ml
    of MGSO4. (This solution is made by dissolving 40,66 g of MgCl2
    (Hydrate) and 49,3g MgSO4 (Hydrate) respectively in 90 mL of
    deionized H2O. Adjust the volume of the solution to 100 mL with
    deionized H2O and sterilize by autoclaving for 20 min at 15 psi
    \[1.05 kg/cm2\] on liquid cycle.)
    
    After the SOB medium has been autoclaved, allow it to cool to 60°C or
    less.

4.  Add 20 mL of a sterile 1 M solution of glucose. (This solution is
    made by dissolving 18 g of glucose in 90 mL of deionized H2O. After
    the sugar has dissolved, adjust the volume of the solution to 100 mL
    with deionized H2O and sterilize by passing it through a 0.22-µm
    filter.)

**10% Glycerol**

| Glycerol | Glycerol           |
| -------- | ------------------ |
| ddH2O    | ddH2O              |
| Total:   | 1L of 10% Glycerol |

**Medium for 293T HEK cells (10%):**

| DMEM - Dulbecco's Modified Eagle Medium | 500 ml |
| --------------------------------------- | ------ |
| FCS                                     | 50 ml  |
| L-Glutamin                              | 5ml    |

{{% /accordion %}}


{{% accordion id="ex4" title="Cultivation of E.coli strains" %}}

**Cultivation of *E.coli* strains:**
---

**Freezing of *E.coli* strains:**

1.  Spinned for 10 minutes at 2,500 rpm

2.  Discarded supernatant without disturbing the pellet, while leaving
    ~1ml for every sample

3.  Resuspend the pellet in the remaining supernatant. 

4.  Added 900μl cell culture and 300μl 99% glycerol, to a final
    glycerol concentration of ~25%. 6.

5.  Store in -80 °C freezer

{{% /accordion %}}


{{% accordion id="ex5" title="Preparation of chemically competent bacteria" %}}

**Preparation of chemically competent bacteria:**
---
*Why did we choose this experiment?*

Competent cells have a increased cell permeability. This is an important
characteristic for the following transformation, where the aim is to
transfer DNA into the bacteria.

**Procedure:**

1. Inoculate 5 ml LB medium with the appropriate antibiotics with the
E. coli strain of which you want to make competent cells and incubate
overnight at 37°C. When preparing DH5a competent cells it is better to
use SOB medium instead of LB.

2. Use the overnight culture to inoculate 500 ml LB medium and incubate
at 30°C until the absorbance at 600 nm is between 0.4-0.6. Optional: add
2.5 ml 2M MgCl2 to the medium (to a final concentration of 10 mM) at the
start of the cultivation.

3. Chill the culture for at least 10 min on ice. In the following steps
the cell suspension should be kept on ice as much as possible.

4. Spin the cell suspension for 10 min at 6000 rpm (Sorvall GSA rotor)
or 4000 rpm when harvested in 50-ml Falcon tubes.

5. Gently resuspend the pellet in 100 ml ice-cold CC buffer in 50-ml
Falcon tubes. Resuspend with a 10-ml serological pipette and avoid
introducing bubbles.

6. Incubate the cell suspension on ice for at least 10 min.

7. Spin for 10 min at 4000 rpm at 4°C.

8. Gently resuspend the pellet in 18.6 ml ice-cold CC buffer and add 1.4 ml DMSO.

9. Incubate the cell suspension on ice for at least 10 min.

10. Distribute the cell suspension in 100-200 µl aliquots in 0.5 or
1.5-ml microfuge tubes.

11. Flash freeze the cell suspension in liquid nitrogen and store the
tubes at -80°C. At -80°C the cell will be competent for at least 6
months.

{{% /accordion %}}


{{% accordion id="ex6" title="Preparation of electro-competent cells" %}}

**Preparation of electro-competent cells**
---

*Why did we choose this experiment?*

Competent cells have a increased cell permeability. This is an important
characteristic for the following transformation, where the aim is to
transfer DNA into the bacteria.

**Prepare first:**

  - 2 liter of LB without NaCl (10 g tryptone, 5 g yeast extract)

  - 250 ml of 8.7 % v/v glycerol autoclaved/sterile

  - 2 liter of milliQ water autoclaved/sterile

  - 250 ml cylinder closed with aluminum foil containing
    autoclaved/sterile

  - 4x 2 L Erlenmeyer flasks autoclaved/sterile

  - 4x 250 ml centrifuge tubes for Beckam rotor J14 autoclaved/sterile

  - 5 ml pipet tips autoclaved/sterile

  - 1 ml pipet tips autoclaved/sterile

  - 200 µl pipet tips autoclaved/sterile

  - At least 500x 0.5 ml eppendorf tubes autoclaved/sterile

  - Required: Centrifuge Beckam with rotor J14 or similar

  - Work very sterile because we do not use any antibiotics\!

**Procedures:**

1.  Reserve 15 ml LB in a small bottle at 4 °C

2.  Inoculate 2 L LB Medium with a starter culture and mix well

3.  Spread to 4x 2 L Erlenmeyer flasks

4.  Incubate at 37 °C shaking 200 rpm

5.  Measure OD 600 of all four flasks every 45 min

6.  If the first flask reaches OD 0.40 put all flasks for 15 min on an
    ice-water bath

7.  Pellet (1.) at 3500 G for 20 min at 4 °C

8.  Discard supernatant and resuspend in 6x 150 ml ice cold water

9.  Pellet (2.) at 5000 G for 20 min at 4 °C

10. Discard supernatant and resuspend in 6x 150 ml ice cold water

11. Pellet (3.) at 5000 G for 20 min at 4 °C

12. Discard supernatant and resuspend in 2x 150 ml ice cold water  
    Attention: 3 buckets are now pooled in one\!

13. Pellet (4.) at 5000 G for 20 min at 4 °C

14. Discard supernatant and resuspend in 2x 100 ml ice cold 8.7 %
    glycerol  
    Attention: use now 8.7 % glycerol, not water\!

15. Pellet (5.) at 5000 G for 20 min at 4 °C

16. Discard supernatant and resuspend in 2x 2.5 ml ice cold 8.7 %
    glycerol

17. Pool both buckets in one 50 ml tube

18. Make 50 µl aliquots in 0.5 ml eppendorf tubes

19. Freeze and store at -80 °C

{{% /accordion %}}


{{% accordion id="ex7" title="Polymerase chain reaction PCR" %}}

**Polymerase chain reaction (PCR):**
---
*Why did we use this method?*

Within the method of PCR it is possible to amplify small fragments of
DNA which is very useful for further investigation. Furthermore, one
task of the project was to perform Gibson Assembly and thereby it was
necessary to fuse overhangs to the DNA fragments. Another method for
cloning was the restriction enzyme digest. Therefore, suitable
restriction sites had to be added to the ends of our DNA fragments. All
these tasks were performed by different programs of PCR using the
Phusion High-Fidelity PCR master mix.

**PCR Mix:**

| Template DNA         | 5µl (250ng)    |
| -------------------- | -------------- |
| Forward primer       | 1.25µl (0.5µM) |
| Reverse primer       | 1.25µl (0.5µM) |
| 2xPhusion Master Mix | 12.5µl         |
| Autoclaved H2O       | 25µl           |

**Distinct programs for PCR:**

*Amplification of DNA constructs:*

| Cycle Step           | Temperature \[°C\] | Time \[min.\] | Cycles |
| -------------------- | ------------------ | ------------- | ------ |
| Initial Denaturation | 98                 | 00:30         | X1     |
| Denaturation         | 98                 | 00:10         | X40    |
| Annealing            | 56.0               | 00:30         |        |
| Extension            | 72.0               | 00:30         |        |
| Final Extension      | 72.0               | 2:00          | X1     |
|                      | 4.0                | hold          |        |

*Adding of restriction sites with PCR:*

| Cycle Step           | Temperature \[°C\] | Time \[min.\] | Cycles |
| -------------------- | ------------------ | ------------- | ------ |
| Initial Denaturation | 98                 | 5:00          | X1     |
| Denaturation         | 98                 | 00:10         | X40    |
| Annealing            | 53.7               | 00:30         |        |
| Extension            | 72.0               | 1:00          |        |
| Final Extension      | 72.0               | 10:00         | X1     |
|                      | 4.0                | hold          |        |

*Adding overhangs for Gibson Assembly:*

| Cycle Step           | Temperature \[°C\] | Time \[min.\] | Cycles |
| -------------------- | ------------------ | ------------- | ------ |
| Initial Denaturation | 98                 | 3:00          | X1     |
| Denaturation         | 98                 | 00:10         | X40    |
| Annealing            | 59.8               | 00:30         |        |
| Extension            | 72.0               | 1:00          |        |
| Final Extension      | 72.0               | 10:00         | X1     |
|                      | 4.0                | hold          |        |

{{% /accordion %}}


{{% accordion id="ex8" title="Restriction digest" %}}

**Restriction digest**
---

*Why did we choose this experiment?*


We used a restriction digest for our cloning strategy.

**Materials:**

\- Template DNA

\- Restriction Enzyme of choice

\- 10x Cutsmart Buffer (NEB)

\- ddH2O Procedure

**Procedure:**

1\. Mix the following reagents.

| Concentration | Reagent               |
| ------------- | --------------------- |
| 4-5µg         | Template DNA          |
| 1µl           | Restriction enzyme(s) |
| 4µl           | Cutsmart Buffer (NEB) |
| Fill to 40µl  | ddH<sub>2</sub>O      |

2\. Incubate 60 minutes 37°C. (respectively to enzyme)

3\. Inactivate 20 minutes at 65°C

{{% /accordion %}}


{{% accordion id="ex9" title="Dephosphoryllation of the digested DNA fragments" %}}

**Dephosphoryllation of the digested DNA fragments**
---


*Why did we choose this experiment?*

After the digestion of the DNA fragments, there is a likelihood that
they relegate again. To prohibit that issue, the method of
dephosphorylation was chosen, were phosphate groups are split of.

**Master Mix:   **

| Name                   | Amount  |
| ---------------------- | ------- |
| DNA                    | 3.3 µl  |
| CutSmart x10 NEB       | 5 µl    |
| Arctic Buffer 10x NEB  | 5 µl    |
| Arctic Phosphatase NEB | 1 µl    |
| SmaI NEB               | 0.5 µl  |
| H2O to 50 µl           | 35.2 µl |

**Procedure:**

1.  Incubate the samples at 37°C for 1 h;

2.  Inactivate at 80  °C for 20 min (80 °C necessary to inactivate the
    phosphatase, SmaI itself 65 °C for 20 min would be sufficient)

{{% /accordion %}}


{{% accordion id="ex10" title="Electroporation" %}}

**Electroporation**
---

*Why did we choose this method?*

This method makes cells permeable which allows to transform DNA into the
cells. The permeabilization is done by short pulses

**Procedure:**

1.  Prepare 17 mm x 100 mm round-bottom culture tubes (e.g. VWR
    \#60818-667) at room temperature. Place SOC recovery medium in a
    37°C water bath. Pre-warm selective plates at 37°C for 1 hour.

2.  Place electroporation cuvettes (1 mm) and microcentrifuge tubes on
    ice.

3.  As a positive control for transformation, dilute the control pUC19
    by 1:5 to a final concentration of 10 pg/μl using sterile water.
    Heat-denatured ligation reactions can be used for electroporation
    directly; however, column purification is recommended.

4.  Thaw NEB Turbo Electrocompetent cells on ice (about 10 min) and mix
    cells by flicking gently. Transfer 25 μl of the cells (or the amount
    specified for the cuvettes) to a chilled microcentrifuge tube. Add 1
    μl of the DNA solution.

5.  Carefully transfer the cell/DNA mix into a chilled cuvette without
    introducing bubbles and make sure that the cells deposit across the
    bottom of the cuvette. Electroporate using the following conditions
    for BTX ECM 630 and Bio-Rad GenePulser electroporators: 2.1 kV,  
    100 Ω, and 25 μF. The typical time constant is \~2.6 milliseconds.

6.  Immediately add 975 µl of 37°C SOC to the cuvette, gently mix up and
    down twice, then transfer to the 17 mm x 100 mm round-bottom culture
    tube.

7.  Shake vigorously (250 rpm) or rotate at 37°C for 1 hour.

8.  Dilute the cells as appropriate then spread 100-200 μl cells onto a
    pre-warmed selective plate.

9.  Incubate plates 8 hours to overnight at 37°C.

{{% /accordion %}}


{{% accordion id="ex11" title="Heat Shock" %}}

**Heat Shock**
---

*Why did we choose this method?*

We have produced chemically competent cells and transformed them with
the method of heat shock. Thereby, we transferred our DNA into the
bacteria with the help of this method.

**Procedure:**

1.  Thaw competent cells on ice.

2.  Chill approximately 5 ng (2 μl) of the ligation mixture in a 1.5 ml
    microcentrifuge tube.

3.  Add 50 µl of competent cells to the DNA. Mix gently by pipetting up
    and down or flicking the tube 4–5 times to mix the cells and DNA. Do
    not vortex.

4.  Place the mixture on ice for 30 minutes. Do not mix.

5.  Heat shock at 42°C for 30 seconds. Do not mix.

6.  Add 950 µl of room temperature media to the tube.

7.  Place tube at 37°C for 60 minutes. Shake vigorously (250 rpm) or
    rotate.

8.  Warm selection plates to 37°C.

9.  Spread 50–100 µl of the cells and ligation mixture onto the plates.

10. Incubate overnight at 37°C.

{{% /accordion %}}


{{% accordion id="ex12" title="Isolation of DNA from transformed bacteria (Mini – Midi Prep)" %}}

**Isolation of DNA from transformed bacteria (Mini – Midi Prep)**
---


*Why did we chose this experiment?*

This experiment enables to isolate DNA out of cells. With this it is
possible to check wether the previous transformation has worked or not.

**Procedure:**

**Mini Prep:**

1. Pick a single colony from a freshly streaked selective plate and
inoculate a starter culture of 2–5ml LB medium containing the
appropriate selective antibiotic. Incubate for approximately 8 h at 37°C
with vigorous shaking (approx. 300 rpm). Use a tube or flask with a
volume of at least 4 times the volume of the culture.

2. Dilute the starter culture 1/500 to 1/1000 into 3 ml selective LB
medium. Grow at 37°C for 12–16 h with vigorous shaking (approx. 300
rpm). Use a flask or vessel with a volume of at least 4 times the volume
of the culture. The culture should reach a cell density of approximately
3–4 x 109 cells per ml, which typically corresponds to a pellet wet
weight of approximately 3 g/liter.

3. Harvest the bacterial cells by centrifugation at 6000 x g for 15 min
at 4°C.
Ifyouwishtostoptheprotocolandcontinuelater,freezethecellpelletsat–20°C.

4. Resuspend the bacterial pellet in 0.3 ml of Buffer P1. Ensure that
RNase A has been added to Buffer P1.

5. Add 0.3 ml of Buffer P2, mix thoroughly by vigorously inverting the
sealed tube 4–6 times, and incubate at

6. Add 0.3 ml of chilled Buffer P3, mix immediately and thoroughly by
vigorously inverting 4–6 times, and incubate on ice for 5 min.
Precipitation is enhanced by using chilled Buffer P3 and incubating on
ice. After addition of Buffer P3, a fluffy white material forms and the
lysate becomes less viscous.

7. Centrifuge at maximum speed in a microcentrifuge for 10 min. Remove
supernatant containing plasmid DNA promptly. Before loading the
centrifuge, the sample should be mixed again. Centrifugation should be
performed at maximum speed in 1.5 ml or 2 ml microcentrifuge tubes
(e.g., 10,000–13,000 rpm in a microcentrifuge).

8. Equilibrate a QIAGEN-tip 20 by applying 1 ml Buffer QBT, and allow
the column to empty by gravity flow.

9. Apply the supernatant from step 7 to the QIAGEN-tip 20 and allow it
to enter the resin by gravity flow.

10. Wash the QIAGEN-tip 20 with 2 x 2 ml Buffer QC. Allow Buffer QC to
move through the QIAGEN-tip by gravity flow.

11. Elute DNA with 0.8 ml Buffer QF. Collect the eluate in a 1.5 ml or
2 ml microcentrifuge tubes (not supplied).

12. Precipitate DNA by adding 0.7 volumes (0.56 ml per 0.8 ml of
elution volume) of room-temperature isopropanol to the eluted DNA. Mix
and centrifuge immediately at≥15,000xg rpm for 30 min in a
microcentrifuge. Carefully decant the supernatant.

13. Wash DNA pellet with 1ml of 70 % ethanol and centrifuge at 15,000xg
for 10min. Carefully decant the supernatant without disturbing the
pellet.

14. Air-dry the pellet for 5–10min, and redissolve the DNA in a
suitable volume of buffer (e.g., TE buffer, pH 8.0, or 10mM Tris·Cl, pH
8.5)

**Midi-Prep:**

1. Harvest overnight bacterial culture by centrifuging at 6000 x g for
15 min at 4°C.

2. Resuspend the bacterial pellet in 4 ml Buffer P1. 

3. Add 4 ml Buffer P2, mix thoroughly by vigorously inverting 4–6 times
and incubate at room temperature (15–25°C) for 5 min. If using LyseBlue
reagent, the solution will turn blue. 

4. Add 4 ml Buffer P3, mix thoroughly by vigorously inverting 4–6
times. Incubate on ice for 15 min. If using LyseBlue reagent, mix the
solution until it is colorless. 

5. Centrifuge at ≥20,000 x g for 30 min at 4°C. Re-centrifuge the
supernatant at ≥20,000 x g for 15 min at 4°C. 6. Equilibrate a
QIAGEN-tip 100 by applying 4 ml Buffer QBT, and allow column to empty by
gravity flow. 

7. Apply the supernatant from step 5 to the QIAGEN-tip and allow it to
enter the resin by gravity flow. 

8. Wash the QIAGEN-tip with 2 x 10 ml Buffer QC. Allow Buffer QC to
move through the QIAGEN-tip by gravity flow. 

9. Elute DNA with 5 ml Buffer QF into a clean 15 ml vessel. For
constructs larger than 45 kb, prewarming the elution buffer to 65°C may
help to increase the yield. 

10. Precipitate DNA by adding 3.5 ml roomtemperature isopropanol to the
eluted DNA and mix. Centrifuge at ≥15,000 x g for 30 min at 4°C.
Carefully decant the supernatant. 

11. Wash the DNA pellet with 2 ml room-temperature 70% ethanol and
centrifuge at ≥15,000 x g for 10 min. Carefully decant supernatant. 

12. Air-dry pellet for 5–10 min and redissolve DNA in a suitable volume
of appropriate buffer (e.g., TE buffer, pH 8.0, or 10 mM Tris·Cl, pH
8.5).

{{% /accordion %}}


{{% accordion id="ex13" title="Colony PCR" %}}

**Colony PCR:**
---
 
*Why did we choose this experiment?*

Polymerase chain reaction is an in vitro technique to exponentially
amplify DNA of interest. Here, we exploit this technique to confirm the
insertion of a specific DNA part into the genome of E. Coli.

**Materials:**

- Template DNA

- Forward Primer

- Reverse Primer

- OneTaq Polymerase

- LB-Agar plate

**Procedure:**

1. Pick a clone with the pipet.

2\. Putting the pipette in the PCR tube and then into the LB with
antibiotics.

3. Mix the following reagents.

4. Run a PCR (see protocol PCR)

| Concentration          | Reagents          |
| ---------------------- | ----------------- |
| 5µl                    | OneTaq buffer     |
| 0.25µM                 | Forward Primer    |
| 0.25µM                 | Reverse Primer    |
| 0.125µl                | OneTaq Polymerase |
| Bacteria from 1 colony | Template          |
| Fill to 25µl           | ddH2O             |

{{% /accordion %}}


{{% accordion id="ex14" title="Agarose-Gel electrophoresis" %}}

**Agarose-Gel electrophoresis**
---

*Why did we choose this experiment?*

This method is used as a quality control of enzymatic reactions on DNA
or RNA. It is also useful for the separation of DNA or RNA fragments of
different lengths.

**Materials**

- DNA of Interest

- Gel Ladder

- 6x purple loading dye

- Agarose

- 1x TAE buffer

- Gel chamber

- Sybr Safe DNA stain

- UV illuminator + camera Procedure

1. Prepare an agarose gel with an appropriate concentration for the
fragment (0.5%-3% (w/v)) in TAE buffer.

2. Heat solution until it is fully dissolved.

3. Add DNA stain.

4. Cast the gel in a gel chamber, add an appropriate comb and wait at
least 20 minutes until the gel is fully polymerized.

5. Mix at least 100ng DNA with loading dye.

6. Load the gel with your sample.

7. Let it run at 120V, 400mA for 20 minutes.

8. Image the gel under a Camera with UV illuminator.

{{% /accordion %}}


{{% accordion id="ex15" title="Gibson Assembly" %}}

**Gibson Assembly**
---
*Why did we choose this method?*

Gibson assembly is a very useful tool for cloning, because it is
possible to fuse several fragments.

**Procedure:**

1.  Set up the following reaction on ice:

|                                 | Rcommended Amount of Fragments Used for Assembly |                    |                  |
| ------------------------------- | ------------------------------------------------ | ------------------ | ---------------- |
|                                 | 2-3 fragments                                    | 4-6 fragments      | Positive control |
| Total Amount of Fragments       | 0.02-0.05 pmol \* x µl                           | 0.2-1 pmol \* x µl | 10 µl            |
| Gibson Assembly Master Mix (2x) | 10 µl                                            | 10 µl              | 10 µl            |
| Deionized H2O                   | 10-X µl                                          | 10-X µl            | 0                |
| Total volume:                   | 20 µl                                            | 20 µl              | 20 µl            |

2.  assembled or 60 minutes when 4-6 fragments are being assembled.
    Following incubation, store samples on ice or at –20°C for
    subsequent transformation.

{{% /accordion %}}


{{% accordion id="ex16" title="Ligation" %}}

**Ligation:**
---

*Why did we choose this experiment?*

With the help of the T4 Ligase, it is possible to connect DNA fragments
which were previous digested by restriction digestion.

**Procedure:**

1.  Set up the following reaction in a microcentrifuge tube on ice.

| Component                  | 20µl reaction        |
| -------------------------- | -------------------- |
| T4 DNA Ligase Buffer (10x) | 2 µl                 |
| Vector DNA (4kb)           | 50 ng (0,020 pmol)   |
| Insert DNA (1kb)           | 37.5 ng (0.060 pmol) |
| Nuclease-free water        | To 20µl              |
| T4 DNA Ligase              | 1 µl                 |

2.  The T4 DNA Ligase Buffer should be thawed and resuspended at room
    temperature.

3.  Gently mix the reaction by pipetting up and down and microfuge
    briefly.

4.  For cohesive (sticky) ends, incubate at 16°C overnight or room
    temperature for 10 minutes.

5.  For blunt ends or single base overhangs, incubate at 16°C
    overnight or room temperature for 2 hours (alternatively, high
    concentration T4 DNA Ligase can be used in a 10 minute ligation).

6.  Heat inactivate at 65°C for 10 minutes.

7.  Chill on ice and transform 1-5 μl of the reaction into 50 μl
    competent cells.
   
 {{% /accordion %}}
 
 
 
 
 {{% accordion id="ex17" title="Gelextraction" %}}
 **Gelextraction**
---

*Why did we choose this experiment?*


We were not interested of the whole sample but just parts of it.
Thereby, with the help of a gelextraction it was possible for us to
investigate just the parts of interest of our samples.

**Materials:**

QIAquick® Gel Extraction Kit

**Procedure:**

1. Excise the DNA fragment from the agarose gel with a clean, sharp
scalpel.

2. Weigh the gel slice in a colorless tube. Add 3 volumes Buffer QG to
1 volume gel (100 mg \~ 100 μl). For \>2% agarose gels, add 6 volumes
Buffer QG.

3. Incubate at 50°C for 10 min (or until the gel slice has completely
dissolved). Vortex the tube every 2–3 min to help dissolve gel.

4. After the gel slice has dissolved completely, check that the color
of the mixture is yellow (similar to Buffer QG without dissolved
agarose)

5. Add 1 gel volume of isopropanol to the sample and mix.

6. Place a QIAquick spin column in a provided 2 ml collection tube

7. To bind DNA, apply the sample to the QIAquick column and centrifuge
for 1 min until all the samples have passed through the column. Discard
flow-through and place the QIAquick column back into the same tube.

8. If the DNA will subsequently be used for sequencing, in vitro
transcription, or microinjection, add 0.5 ml Buffer QG to the QIAquick
column and S centrifuge for 1 min. Discard flow-through and place the
QIAquick column back into the same tube.

9. To wash, add 0.75 ml Buffer PE to QIAquick column and centrifuge for
1 min. Discard flow-through and place the QIAquick column back into the
same tube.

10. Centrifuge the QIAquick column once more in the provided 2 ml
collection tube for 1 min at 17,900 x g (13,000 rpm) to remove residual
wash buffer.

11. Place QIAquick column into a clean 1.5 ml microcentrifuge tube.

12. To elute DNA, add 50 μl water to the center of the QIAquick
membrane and centrifuge the column for 1 min.

13. If the purified DNA is to be analyzed on a gel, add 1 volume of
Loading Dye to 5 volumes of purified DNA. Mix the solution by pipetting
up and down before loading the gel.
   
 {{% /accordion %}}


 {{% accordion id="ex18" title="Cultivation of HEK 293T cells" %}}
 
 **Cultivation of HEK 293T cells:**
---

*Why did we choose HEK 293T cells?*

These cells functioned as an expression system, to produce our proteins
of interest.

**Procedure:**

**Apply frozen 293T cells to cell culture**

1.  Fill 10ml medium into a 75 cm² culture flask

2.  Frozen cells in cryotubes should be defrosted quickly under e.g.
    warm water.

3.  Apply cells to centrifuge tubes and with 5ml medium

4.  Centrifuge the tubes at 900 U for 5 min and discard the supernatant

5.  Resuspend the pellet in 5ml medium and put the suspension the the
    prepared culture flask from step 1

**Change medium of the adherent 293T cells**

1.  Discard medium

2.  Add new medium into the culture flask

**Trypsinization of the 293T cells to detach them from the culture flask
wall**

 1. Discard medium
 
2. Washing with 3 ml trypsin
 
3. Incubation with 3 ml trypsin

4. When cells are detached mixing with 7 ml Medium 
 
5. Counting in neubauer chamber
 
6. Add respective amount of medium
   
 {{% /accordion %}}


 {{% accordion id="ex19" title="FACS (fluorescence activated cell sorting)" %}}
 
 **FACS (fluorescence activated cell sorting):**
---

*Why did we choose this experiment?*

With the FACS method, it is possible to send cells trough a laser sensor
and check the condition of the cells.

**Procedure:**

1.  Overnight culture of the bacteria

2.  For prohibiting clumbing of the cells, they were treated 1:1 with
    ELISA buffer (with 0,05% Tween) direct before the measurement

3.  Cells (*E.coli* bacteria) were not stained before FACS analysis

4.  Cells were checked for fluorescence to measure CFP expression
    
    1.  Excitation at 405nm
    
    2.  Detection of CFP at 485nm
    
    3.  Gain: FSC 165, SSC 400
   
 {{% /accordion %}}


 {{% accordion id="ex20" title="Protein purification by His SpinTrap" %}}
 
 **Protein purification by His SpinTrap**
---

*Why did we choose this experiment?*

His-tag purification uses the purification technique of immobilized
metal affinity chromatography. In this technique, transition metal ions
are immobilized on a resin matrix using a chelating agent such as
iminodiacetic acid. The most common ion for the his-purification is
Ni2+. The His-tag has high affinity for these metal ions and binds
strongly to the column. Imidazole competes with the His-tag for binding
to the metal-charged resin and thus is used for elution of the protein

Binding buffer: 50mM NaH2PO4, 500 mM NaCl, 20 mM Imidazol, pH 8,0

Elution buffer: 50 mM NaH2PO4, pH 8,0, 500 mM NaCl, 500 mM Imidazol

**Procedure:**

1.  Place the column in a 2 ml microcentrifuge tube to collect the
    liquid during centrifugation.

2.  Remove storage solution

- Invert and shake the column repeatedly to resuspend the medium.

- Loosen the top cap one-quarter of a turn and twist off the bottom
closure.

- Place the column in a 2 ml microcentrifuge tube and centrifuge for
30 s at 70 to 100 × g.

- Remove and discard the top cap.

3.  Column equilibration

- Add 600 µl binding buffer.

- Centrifuge for 30 s at 70 to 100 × g

4.  Sample application

- Add up to 600 µl sample in one application. • Centrifuge for 30 s at
 70 to 100 × g.

5.  Wash

- Add 600 µl binding buffer.

- Centrifuge for 30 s at 70 to 100 × g.

6.  Elution

- Add 200 µl elution buffer.
 
- Centrifuge for 30 s at 70 to 100 × g and collect the purified
sample.

- Add 200 µl elution buffer.

- Centrifuge for 30 s at 70 to 100 × g and collect the purified sample
   
 {{% /accordion %}}


 {{% accordion id="ex21" title="Western Blotting" %}}
 
 **Western Blotting**
---

*Why we chose this experiment*

Different purification tags are commonly used to isolate fusion proteins
or to detect certain proteins within lysates or other protein mixtures.
Western Blotting is a technique used to identify small amount s of
tagged proteins. Using a primary antibody to detect the specific target
or protein present in the sample and a secondary antibody specifically
detecting the bound primary antibody, Western blots are a very common
tool to determine the presence of the target proteins within a sample.

**Materials**

- Anti-6xHis Antibody (mouse)

- HRP anti-mouse (donkey)

- Washing buffer (1xPBS, 0.1% Tween-20)

- Transfer Buffer

- PVDF blotting membrane

- Blocking Buffer (10% milkpowder solution)

- Filter paper

- Methanol

- Proteinmarker: Page ruler: Protein Ladder Thermo Fisher

**Procedure**

1. Preperation of a 10% Bis-tris Gel. 

2. Mixing of 10 µl harvested cells with 2x loading buffer

3. 95°C for 5 min

4. Loading of the gel with 5µL marker

5. Running gel for 1 h at 130V in 1x MOPS buffer

6. Blotting to the membrane at 65mA for 1h30

7. Blocking membrane in 10% milkpowder for 1h30

8. Washing with PBST 

9. Adding primary antibody (mouse anti-His in 1:2000 dilution) 10mL in
10% milkpowder

10. Shaking overnight at 4°C

11. 3 times washing for 10 min with PBST

12. Adding secondary antibody (HRP anti mouse in 1:2000 dilution) 10 ml
in 10% milkpowder

13. Shaking at RT for 1h

14. 3 times washing for 10 min with PBST

15. Adding 2mL of the reaction mix directly before taking the picture
of the blot


----

For the picture select channel "Chemiluminescence" to detect the bands.

For the marker select "Copy Marker" and don’t move the membrane between
the two pictures\!
   
 {{% /accordion %}}
















